DiscordBot.DataAccess/EventsSheetsService.cs

```diff
@@ -40,527 +41,153 @@ public interface IEventsSheetsService
         Task<Dictionary<EventResponse, IEnumerable<ulong>>> GetSignupsByResponseAsync(int eventId);
     }
 
-    public class EventsSheetsService : IEventsSheetsService
+    public class EventsSheetsService : UnsafeEventsSheetsService
```

Comment:
I don't think an inheritance relationship is quite what you want here. Because you're not extending the functionality of the unsafe class, instead you're providing a wrapper for it. So I would recommend:
* Have both classes implement the sheets service interface. This will enforce the wrapper class implementing all the same methods as the inner class.
* Have the wrapper class contain an instance of the inner class which it will call methods on when safe
* Change the visibility of the inner class so that it can't be instantiated outside of this project

What do you think? 🙂

Reply:
Yeah I agree. I had considering both of these approaches and was a bit unsure of which to use. I probably should have checked that with you, but at least it didn't take me too long to switch it over. Changed to a wrapper class in 81c8e0e 🙂